clang/lib/Sema/SemaSPIRV.cpp

```diff
@@ -16,88 +16,90 @@ namespace clang {
 
 SemaSPIRV::SemaSPIRV(Sema &S) : SemaBase(S) {}
 
+/// Checks if the first `NumArgsToCheck` arguments of a function call are of
+/// vector type. If any of the arguments is not a vector type, it emits a
+/// diagnostic error and returns `true`. Otherwise, it returns `false`.
+///
+/// \param TheCall The function call expression to check.
+/// \param NumArgsToCheck The number of arguments to check for vector type.
+/// \return `true` if any of the arguments is not a vector type, `false`
+/// otherwise.
+
+bool SemaSPIRV::CheckVectorArgs(CallExpr *TheCall, unsigned NumArgsToCheck) {
```

Comment:
I suspect most cases all the function arguments will be the same type. creae an overload of this function that does not take a `NumArgsToCheck` and instead just calls  this function:
```suggestion
bool SemaSPIRV::CheckVectorArgs(CallExpr *TheCall) {
  return CheckVectorArgs(theCall, TheCall->getNumArgs());
}
bool SemaSPIRV::CheckVectorArgs(CallExpr *TheCall, unsigned NumArgsToCheck) {
```

`reflect` and `distance` should use the one that doesn't require you to specify argument size.